app/repositories/SessionRepository.scala

```diff
@@ -17,116 +17,81 @@
+      IndexModel(
+        Indexes.ascending("internalId"),
+        IndexOptions().name("internal-auth-id-index")
+      )
+    )
   )
+    with Logging with SessionRepository {
 
-  private val internalAuthIdIndex = MongoIndex(
-    key = Seq("internalId" -> IndexType.Ascending),
-    name = "internal-auth-id-index"
-  )
+  def get(id: String): Future[Option[UserAnswers]] = {
 
-  override def get(id: String): Future[Option[UserAnswers]] = {
-    val selector = Json.obj(
-      "internalId" -> id
-    )
+    val selector = equal("internalId", id)
 
-    val modifier = Json.obj(
-      "$set" -> Json.obj(
-        "updatedAt" -> MongoDateTimeFormats.localDateTimeWrite.writes(LocalDateTime.now)
-      )
-    )
+    val modifier = Updates.set("updatedAt", LocalDateTime.now())
 
-    collection.flatMap {
-      _.findAndUpdate(
-        selector = selector,
-        update = modifier,
-        fetchNewObject = true,
-        upsert = false,
-        sort = None,
-        fields = None,
-        bypassDocumentValidation = false,
-        writeConcern = WriteConcern.Default,
-        maxTime = None,
-        collation = None,
-        arrayFilters = Nil
-      ).map(_.result[UserAnswers])
-    }
-  }
+    val updateOption = new FindOneAndUpdateOptions().upsert(false)
 
-  override def set(userAnswers: UserAnswers): Future[Boolean] = {
+    collection.findOneAndUpdate(selector, modifier, updateOption).toFutureOption()
 
-    val selector = Json.obj(
-      "internalId" -> userAnswers.id
-    )
+  }
 
-    val modifier = Json.obj(
-      "$set" -> (userAnswers copy (lastUpdated = LocalDateTime.now))
-    )
+  def set(userAnswers: UserAnswers): Future[Boolean] = {
 
-    collection.flatMap {
-      _.update(ordered = false).one(selector, modifier, upsert = true, multi = false).map {
-        result => result.ok
-      }
-    }
+    val selector = equal("internalId", userAnswers.id)
+
+    collection.replaceOne(selector, userAnswers.copy(lastUpdated = LocalDateTime.now), ReplaceOptions().upsert(true))
+      .head()
+      .map(_.wasAcknowledged())
   }
 
-  override def resetCache(internalId: String): Future[Option[JsObject]] = {
+  def resetCache(internalId: String): Future[Option[Boolean]] = {
 
-    val selector = Json.obj(
-      "internalId" -> internalId
-    )
+    val selector = equal("internalId",internalId)
+
+    collection.deleteOne(selector).toFutureOption().map(_.map(_.wasAcknowledged()))
 
-    collection.flatMap(_.findAndRemove(selector, None, None, WriteConcern.Default, None, None, Seq.empty).map(
-      _.value
-    ))
   }
+
 }
 
+@ImplementedBy(classOf[DefaultSessionRepository])
 trait SessionRepository {
 
   def get(id: String): Future[Option[UserAnswers]]
 
   def set(userAnswers: UserAnswers): Future[Boolean]
 
-  def resetCache(internalId: String): Future[Option[JsObject]]
-}
+  def resetCache(internalId: String): Future[Option[Boolean]]
```

Comment:
why you have Option[Boolean] and not just boolean? you can default it to false if nothing got back from database like here:
```
collection.updateOne(selector, modifier, updateOptions).headOption().map(_.exists(_.wasAcknowledged()))
```
this is just an example how I did in other project your return type might be different so might need some changes

Reply:
this change is still not done

Reply:
done